ui/apps/platform/cypress/integration/violations/violations.test.js

```diff
@@ -154,117 +123,81 @@ describe('Violations page', () => {
+        cy.get(ViolationsPageSelectors.enforcement.explanationMessage).should(
+            'contain',
+            'Runtime data was evaluated against this security policy'
+        );
     });
 
     it('should have deployment information in the Deployment tab', () => {
-        mockGetAlert();
-        cy.get(ViolationsPageSelectors.firstTableRowLink).click();
-        cy.wait('@alertById').then((interception) => {
-            const { deployment } = interception.response.body;
-            cy.intercept('GET', `${api.risks.getDeployment}`, {
-                body: deployment,
-            }).as('deployment');
-            cy.get(ViolationsPageSelectors.details.deploymentTab).click();
-            cy.wait('@deployment');
-
-            cy.get(ViolationsPageSelectors.deployment.overview).should('exist');
-            cy.get(ViolationsPageSelectors.deployment.containerConfiguration).should('exist');
-            cy.get(ViolationsPageSelectors.deployment.securityContext).should('exist');
-            cy.get(ViolationsPageSelectors.deployment.portConfiguration).should('exist');
-
-            // TODO does not exist: should it?
-            // cy.get(ViolationsPageSelectors.deployment.snapshotWarning).should('exist');
-        });
+        visitViolationWithFixture('alerts/alert0.json');
+        clickDeploymentTabWithFixture('alerts/deployment0.json');
+
+        cy.get(ViolationsPageSelectors.deployment.overview);
+        cy.get(ViolationsPageSelectors.deployment.containerConfiguration);
+        cy.get(
+            `${ViolationsPageSelectors.deployment.containerConfiguration} [data-testid="commands"]`
+        ).should('not.exist');
+        cy.get(ViolationsPageSelectors.deployment.securityContext);
+        cy.get(ViolationsPageSelectors.deployment.portConfiguration);
     });
 
     it('should show deployment information in the Deployment Details tab with no container configuration values', () => {
-        mockGetAlertWithEmptyContainerConfig();
-        cy.get(ViolationsPageSelectors.lastTableRowLink).click();
-        cy.wait('@alertWithEmptyContainerConfig');
-        cy.get(ViolationsPageSelectors.details.deploymentTab).click();
+        visitViolationWithFixture('alerts/alertWithEmptyContainerConfig.json');
+        clickDeploymentTabWithFixture('alerts/deploymentWithEmptyContainerConfig.json');
+
+        cy.get(ViolationsPageSelectors.deployment.containerConfiguration);
+        // TODO need more positive and negative assertions to contrast deployments in this and the previous test.
         cy.get(
             `${ViolationsPageSelectors.deployment.containerConfiguration} [data-testid="commands"]`
         ).should('not.exist');
     });
 
     it('should have policy information in the Policy Details tab', () => {
-        mockGetAlert();
-        cy.get(ViolationsPageSelectors.firstTableRowLink).click();
-        cy.wait('@alertById');
+        visitViolationWithFixture('alerts/alert0.json');
```

Comment:
nit: Would be nice to have more descriptive fixture filenames than `alert0`, `deployment0`, etc.

Reply:
Good point. I will rename the files to make clearer that the fixtures for `Alert` and `Deployment` correspond to the `ListAlert` item at index 0 in the alerts.json fixture. The relationship matters only for `'should go to the detail page on row click'` test above.